File: php/muse/util/SequenceBuilder.php
<?php
    declare(strict_types=1);

    namespace muse\util\SequenceBuilder;

    require_once "muse/util/RandomFactory.php";
    require_once "muse/util/Sequences.php";

    use muse\util\RandomFactory;
    use muse\util\Sequences;

    function packIncreasing(array &$arr): void {
        $arr[0] = RandomFactory\genIntN(1, 3);
        for ($i = 1, $length = count($arr); $i < $length; $i++) {
            $arr[$i] = $arr[$i - 1] + RandomFactory\genIntN(1, 3);
        }
    }

    function packRandom(array &$arr): void {
        foreach ($arr as &$v) {
            $v = RandomFactory\genInt();
        }
    }

    function packDecreasing(array &$arr): void {
        packIncreasing($arr);
        Sequences.reverse($arr);
    }
?>

File: python/muse/util/SequenceBuilder.py
from muse.util import RandomFactory, Sequences


def packIncreasing(arr: list) -> None:
    arr[0] = RandomFactory.genIntN(1, 3)
    for i in range(1, len(arr)):
        arr[i] = arr[i - 1] + RandomFactory.genIntN(1, 3)


def packRandom(arr: list) -> None:
    for i, _ in enumerate(arr):
        arr[i] = RandomFactory.genInt()


def packDecreasing(arr: list) -> None:
    packIncreasing(arr)
    Sequences.reverse(arr)
